Revisions after reviewer comments

mjhinch opened the pull request:
--Removed an extra `)`from fcs_processing() chunk in the `README`
--Added code demonstrating how to install from Bioconductor
--Added explicit description of default parameters for the `quality_report()` function.

mjhinch committed:
Updated installation instructions

-Added code installation instructions for installing from bioconductor
-Updated text defining differences between github and bioconductor version installation

mjhinch committed:
Removed comma from fcs_processing chunk

-Removed extra comma from the end of the function

mjhinch committed:
Explicitly stating default parameters

- Added explanation of default parameters with recommended values for macaca mulatta, mus musculus, and uncertain experiments

mjhinch committed:
Added "to" in comment for `compensation` argument

-Added a suspected missing word to the description of "compensation" argument.

rodrigarc commented on README.md:
```diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+# `trim_start` = 50 
+# `trim_finish` = 409
+# `trimmed_mean_quality` = 30
+# `posvalue_probe1` = 600 
```
remove the `posvalue_probe1` and `posvalue_probe2`, we should change the default later on since this parameter is very experiment specific

mjhinch replied:
Addressed in c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
````diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+Default parameters for T cell receptors in mice, mus musculus: 
+
+```r
+# `trim_start` = 200 
````
The changed defaults are raw_length and trim_end, not trim_start. 
raw_length = 200
trim_end = 250

mjhinch replied:
Changes made in commit c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
````diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+```r
+# `trim_start` = 200 
+# `trim_finish` = 250
+# `trimmed_mean_quality` = 30
````
you can remove the trimmed_mean_quality

mjhinch replied:
Addressed in commit  c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
````diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
 ```
+## Default parameters:
+
+The default parameters were derived from data sets analyzing B cell receptor sequences from rhesus macaques, macaca mulatta. These defaults can be used in other experiments or can be changed dependent on the users' needs. Below are potential defaults for B cell receptors for macaques and T cell receptors for mice. For other species not listed or if you are uncertain, a `trimmed_mean_quality` or m ...
````
Could you change the text a bit? 
- italic species names: 
    *Macaca mulatta*
- add the origin of the dataset: 
    "from a data set analyzing B cell receptor sequences from rhesus macaques generated by [Ols et al., *Immunity* 2023](https://zenodo.org/records/7895251). To generate this dataset, primers from [Sundling et al., *Sci. Transl. Med* 2013](https://doi.org/10.1126/scitranslmed.3003752) and [*J. Immunol. Methods* 2013](https://doi.org/10.1016/j.jim.2012.09.003) were used. Those primers were developed to capture the entire V region from heavy and light chain from macaque B cell receptors, the mean length of those sequences was 460 basepairs (bp). The default parameters were also tested for a human B cell receptor dataset generated using primers from [Doria-Rose et al., *J. Virol.* 2016](https://doi.org/10.1128/jvi.01791-15). Additionally, we have tested scifer for gamma delta T cell receptor sequences from human and mice (*Mus musculus*), for those it was required to change the default parameters due to its shorter amplicon length of 250 bp based on the primers used. **The most important information is the length of your expected amplicon, with a different amplicon length you should change the default parameters**, most importantly the `raw_length` and `trim_finish` parameters since they rely on your amplicon length. The default of minimum mean Phred quality score equal to 30 is standard to achieve mean of 99.9% base call accuracy. Below you can find the defaults that are used by scifer and what you should change for other species or amplicons. "

mjhinch replied:
Updated in commit in c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
```diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+
+The default parameters were derived from data sets analyzing B cell receptor sequences from rhesus macaques, macaca mulatta. These defaults can be used in other experiments or can be changed dependent on the users' needs. Below are potential defaults for B cell receptors for macaques and T cell receptors for mice. For other species not listed or if you are uncertain, a `trimmed_mean_quality` or m ...
+
+Default parameters for B cell receptors in macaques: 
```
"in macaques and humans based on the primers mentioned above"

mjhinch replied:
Addressed in c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
````diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+Default parameters for B cell receptors in macaques: 
+
+```r
+# `raw_length` = 400 
````
add a comment for each parameters with a short sentence that explains what is that parameters
```
# `raw_length` = 400 # based on your expected amplicon length
```

rodrigarc replied:
the the raw_length default is 343, not 400, I think you might be checking an older version of scifer installed, or the one from bioconductor which is not updated, check on github

mjhinch replied:
Addressed in c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
````diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+
+```
+
+Default parameters for T cell receptors in mice, mus musculus: 
````
Default parameters for gamma delta T cell receptors in mice (*Mus musculus*):

mjhinch replied:
Addressed in c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc commented on README.md:
````diff
@@ -170,6 +170,41 @@ quality_report(folder_sequences = directory_sequences,
+```
+
+
+Default parameters for Y cell receptors in zebrafish / other species / when uncertain:
````
Change this one to be more broad and add example code **that works** to explain how they can check their  sequences quality:
eg:
If you are uncertain of your amplicon length, first check the quality metrics of your sequences using `summarise_quality`. Save the results in an object to evaluate the quality metrics columns, especially raw.length` and `trim.finish` columns, such as the mean, min, and max values. This will help you to identify which parameters you should change. 

```
example code

```

If you are still uncertain you can use a broad default parameters that will filter mostly on mean base call accuracy. To do that, change the `raw_length` to 0 and and `trim_finish` to 50. This will null the filter based on length but the base mean accuracy  of 99.9% will be maintained. You can also put cdr3_start and cdr3_end to 0 if there is not position that you are particular interested in having with higher accuracy.

```
example code
```

mjhinch replied:
Added in commit: c446337ebf9b5807fadb42ba98e9359ac8cb9b7e

rodrigarc reviewed (changes requested):
@mjhinch I have added comments throughout the commits to make some changes

mjhinch committed:
Updated version

--Updated version to 1.7.3

mjhinch committed:
Revised README for pull request

-Requested changes to descriptions and example code segments

rodrigarc commented:
add the new r-cmd-check to run only on itself and then you can fix the issue of not running the checks before we merge

rodrigarc commented on README.md:
````diff
@@ -172,40 +172,60 @@ quality_report(folder_sequences = directory_sequences,
+
+```r
+quality_control_summary <- summarise_quality(folder_sequences = directory_sequences,
+                                         secondary.peak.ratio = 0.33,
````
remove line 211 and 212

mjhinch replied:
This code chunk runs without error or warnings, addressed in commit 8047b39466702d35773fd07e2e205a2b73dcb09e

mjhinch replied:
8047b39466702d35773fd07e2e205a2b73dcb09e

rodrigarc replied:
this commit is not linked but I will assume it is solved anyway

rodrigarc commented on README.md:
````diff
@@ -172,40 +172,60 @@ quality_report(folder_sequences = directory_sequences,
+For other species not listed or if you are uncertain, a `trimmed_mean_quality` or minimum phred quality score = 30 should be used as this returns an average of 99.9% base call accuracy for the analyzed samples.
 
 ```r
 # `trimmed_mean_quality` = 30
````
this is already the default, so if someone wants to use mainly that, they should change the other parameters, like `raw_length`, not this one

mjhinch replied:
Removed lines 200-204 consistent with comments reflecting `raw_length`, addressed in commit 41f5df08ea0f87c44bef88b01d06e0ee5333ea82

rodrigarc commented on README.md:
```diff
@@ -172,40 +172,60 @@ quality_report(folder_sequences = directory_sequences,
+               outputfile = "QC_report.html",
+               output_dir = "~/Desktop/project_folder/results_output",
+               folder_path_fcs = directory_flowdata,
+               probe1 = "PE.Cy7.A", probe2 = "Alexa.Fluor.700.A",
```
remove 224-225 or change it to something like "probe1 = "my_probe_channel_1"

mjhinch replied:
Addressed in commit 8714bee89b7765008e7b9cde4e08dd037a47482e

rodrigarc commented on README.md:
````diff
@@ -172,40 +172,60 @@ quality_report(folder_sequences = directory_sequences,
 ```r
-# `raw_length` = 400 
+# `raw_length` = 343 # based on your expected amplicon length
 # `trim_start` = 50 
````
add similar comments to all the parameters

mjhinch replied:
Addressed in commit 6bd97d120db2f1209e72cd1ca00641db1dc7e078

mjhinch committed:
Removed lines 211 & 212

--This code runs without error or warning
--removed lines containing arguments (211-212)

mjhinch committed:
Removed explanation of phred score

--Removed chunk with details of phred score as default parameter

mjhinch committed:
Generalized probe1 argument

--Changed argument placeholder to "my_probe_channel_1"

mjhinch committed:
-Added comments to default argument examples

-- Added comments to default parameter examples

mjhinch committed:
Updated introduction to default parameters

- Updated introduction for default parameters: spelling

mjhinch committed:
Added flow cytometry default explanation to README

--Added documentation to README for flow data sets

rodrigarc commented on README.md:
````diff
@@ -208,15 +208,53 @@ If you are still uncertain you can use a broad default parameters that will filt
+#             )
 ```
 
+The default for `compensation` is `FALSE` though often the samples will have been compensated before sorting. Setting `compensation` to `TRUE` makes it so that compensation matrix within the index files will be already automatically applied.
````
"though often the samples will have been compensated before sorting."
"though often the samples were already compensated before sorting"

mjhinch replied:
Changed in e7848a3f4b8050d7acd6b7aa6d60980a68f65fb1

rodrigarc commented on README.md:
````diff
@@ -208,15 +208,53 @@ If you are still uncertain you can use a broad default parameters that will filt
+```
+
+For more information please consult the vignette [here](https://bioconductor.org/packages/release/bioc/vignettes/scifer/inst/doc/flow_cytometry_data.html)
+
````
Add at the end that 

"
At the moment scifer only accepts merging and integrating data with two channels ( eg. 2 probes, one per channel). Furthermore, scifer does not do batch correction for flow cytometry data, thus different experiments should have their Median Fluorescence Intensity (MFI) normalized before running `fcs_processing`.
"

mjhinch replied:
Changed in e7848a3f4b8050d7acd6b7aa6d60980a68f65fb1

rodrigarc commented on README.md:
````diff
@@ -208,15 +208,53 @@ If you are still uncertain you can use a broad default parameters that will filt
+
+```
+
+The `specificity` column is based on your selected channels and their thresholds and uses these thresholds to name your sorted cells. Probed samples  would have a `Pre.F` single-positive, `Post.F` single-positive, double-positive cells named as`DP`, and double-negative cells named as `DN`. Which can be seen using: 
````
could you add this?
"
The value used as input to `posvalue_probe1` and `posvalue_probe1` is the Mean Fluorescence Intensity (MFI) for a given channel. 
"

mjhinch replied:
Addressed in e7848a3f4b8050d7acd6b7aa6d60980a68f65fb1

rodrigarc commented on README.md:
````diff
@@ -208,15 +208,53 @@ If you are still uncertain you can use a broad default parameters that will filt
+fcs_plot(fcs_data)
+```
+
+For more information please consult the vignette [here](https://bioconductor.org/packages/release/bioc/vignettes/scifer/inst/doc/flow_cytometry_data.html)
````
Could you also add that link at the beginning? in the first paragraph already mention this.

mjhinch replied:
The beginning of the default parameters section or at the top?

mjhinch replied:
Addressed in commit fc8f459c84d9195c02cb8bdb4e80357e9dc7ee10

rodrigarc commented on README.md:
```diff
@@ -29,6 +29,8 @@ if (!requireNamespace("BiocManager", quietly = TRUE))
 BiocManager::install(c("DECIPHER","sangerseqR","ape", "BiocStyle"))
```
we need to that a comment about force = TRUE, stating already here that a user might need to set it up as true if older versions of these packages are already installed

mjhinch replied:
Addressed in commit 411c4a41425e204d55aa0b1da80a88023b390b4e

rodrigarc commented:
We have to add that we tested mostly on 4.2 and 4.4 R versions, nothing before 4.0, from what I recall. I do not know where to put at the moment, but something like "scifer was tested mainly in R >= 4.2 versions, so the newest R releases are recommended"

mjhinch committed:
Updated force = true condition for installation

-Adding `force = TRUE` to installation segment if the user has older packages already installed

mjhinch committed:
Added & updated text to the default parameters

-Added addition text to the default parameters describing the arguments within `fcs_processing()`

mjhinch committed:
Added additional walkthrough_link to README

-Added additional reference to the walkthrough vignette

mjhinch committed:
Comment of testing of R versions

-Added comment on which versions of R `scifer` has been tested in

mjhinch committed:
Updated and moved R/Bioc versions in Readme

--Moved R & Bioc versions to above installation and under author and maintainer section. Improved language
